iceoryx_posh/test/fuzztests/README.md

```diff
@@ -0,0 +1,49 @@
+For some fuzzers like the afl, the code needs to be instrumented first. For afl you can download the [LLVM](https://llvm.org/releases/download.html "LLVM"). Afterwards you need to compile the code with an afl compiler which does the instrumentation such as afl-clang-fast.  
+
+Go to your iceoryx folder and change the compiler for example via cmake: 
+
+_cmake -Bbuild -Hiceoryx\_meta -DBUILD\_TEST=True -DCMAKE\_C\_COMPILER=afl-clang-fast -DCMAKE\_CXX\_COMPILER=afl-clang-fast++_
+Afterwards you can build the project for example via:
+_cmake --build build_
+
+Iceoryx is now ready to be fuzzed. 
+
+##How to start testing 
+The fuzz-wrapper binary can be found in \[path\_to\_iceoryx\]/iceoryx\_eclipse/build/posh/test/posh\_fuzztests
+
+To start fuzzing with afl, you need two folders: One which contains examples of valid messages which can be send to the interface and one folder where the fuzzing results are stored. There is already an example folder with a valid text message to test the uds communication. It can be found within \[path\_to\_iceoryx\]/iceoryx\_posh/test/fuzztests/fuzz\_input/ .
+
+The fuzz wrappers currently support three different interfaces: the unix domain socket communication, the message parsing method and the toml configuration file parser.  
+1. The unix domain socket can either be tested by starting RouDi and sending messages to RouDi via uds or by directly invoking the message parsing function within RouDi. The first solution has the advantage, that the messages from uds to RouDi take the same path as normal messages from applications would do. The drawback however is that RouDi needs to be started with all the overhead which would not be necessary for this test which therefore slows the test a little bit.
+
+2. By invoking the message parsing function directly, the method is independent from the underlying protocol such as uds. It should also be slightly faster since some functions are not invoked compared to uds fuzzing. However, a RouDi thread is also started with this approach because otherwise it wouldn't be possible to invoke the message process function within RouDi without directly modifying the code in RouDi. 
+
+3. As a third use-case, the toml configuration parser can be tested. An example of a toml file can be found here: \[path\_to\_iceoryx\]/iceoryx\_posh/etc/iceoryx/roudi\_config\_example.toml. If you chose this method, you also need to set -t or --toml-file <PATH\_TO\_FILE>. A file needs to be specified which can be used to write in the messages which will send to the toml configuration parser. 
+
+The interface you want to fuzz can be chosen with -f or --fuzzing-api + entering the argument: uds, com or toml. The argument uds is used for use-case 1, com for use-case 2 and toml for use-case 3. 
+
+The way how the fuzzer sends messages to the interface can also be specified via -m or --input-mode
+1. The messages can be sent via stdin (-m stdin)
```

Comment:
When i take your example snippet and change -m to `-m foobar` it still start the fuzzer. I saw that `afl-fuzz` is using the same option. I guess there is expected to terminate then but the option is overriden.

Reply:
@dkroenke Thank you for the feedback. 
If you start the fuzzer giving -m foobar as a parameter for the fuzz wrappers, for example with: afl-fuzz -m 1000000 -t 10000 -i ../../../iceoryx_posh/test/fuzztests/fuzz_input/ -o ./fuzz_output ./posh_fuzztests -f uds -m foobar
the fuzzer still works. But what happens is that the fuzz-wrappers exit for every input of the fuzzer because the wrappers exits with the message "No or wrong command lines were specified. Please use --help!"
AFL should show a warning: (odd, check syntax!) because it does not find new paths since the program terminates every time. 
You can verify the behaviour if you execute posh_fuzztest -f uds -m stdin 
You will see that the wrapper terminates every time no matter which input you send via stdin.